File: python/minimum-sum-of-four-digit-number-after-splitting-digits.py
class Solution:
    def minimumSum(self, num: int) -> int:
        h = []
        while num:
            heappush(h, num % 10)
            num //= 10
        result = 0
        while h:
            result *= 10
            result += heappop(h) + heappop(h)
        return result

File: rust/minimum-sum-of-four-digit-number-after-splitting-digits.rs
use std::collections::BinaryHeap;

impl Solution {
    pub fn minimum_sum(mut num: i32) -> i32 {
        let mut heap = BinaryHeap::new();
        let mut result = 0;
        while 0 < num {
            let d = num % 10;
            heap.push(-d);
            num /= 10;
        }
        while !heap.is_empty() {
            result *= 10;
            result -= heap.pop().unwrap() + heap.pop().unwrap();
        }
        return result;
    }
}
